tests/state_migration_test.go

```diff
@@ -55,15 +57,91 @@ func TestMigration_ContinuesRestartAndMigration(t *testing.T) {
 		newPathKey := append([]byte("databaseDirectory"), common.Int64ToByteBigEndian(uint64(database.StateTrieDB))...)
 		newStateTriePath, err := node.ChainDB().GetMiscDB().Get(newPathKey)
 		assert.NoError(t, err)
-		assert.NotEqual(t, stateTriePath, newStateTriePath)
+		assert.NotEqual(t, stateTriePath, newStateTriePath, "migration failed")
 		stateTriePath = newStateTriePath
 	}
 
 	stopNode(t, fullNode)
 }
 
-// if migration status is set on miscDB and a node is restarted, migration should start
+// state trie DB should be determined by the values of miscDB
 func TestMigration_StartMigrationByMiscDB(t *testing.T) {
+	fullNode, node, validator, _, workspace, _, _, _ := newSimpleBlockchain(t, 10)
+	defer os.RemoveAll(workspace)
+
+	stateTriePathKey := append([]byte("databaseDirectory"), common.Int64ToByteBigEndian(uint64(database.StateTrieDB))...)
+
+	// use the default StateTrie DB if it is not set on miscDB
+	{
+		key, err := node.ChainDB().GetMiscDB().Get(stateTriePathKey)
+		assert.Error(t, err)
+		assert.Len(t, key, 0)
+		fullNode, node = checkStatTrieDB(t, fullNode, node, validator, workspace, "statetrie")
+	}
+
+	// after migration, use the created DB
+	{
+		startMigration(t, node)
+		time.Sleep(time.Second)
+		newDBPath2, err := node.ChainDB().GetMiscDB().Get(stateTriePathKey)
+		assert.NoError(t, err)
+		fullNode, node = checkStatTrieDB(t, fullNode, node, validator, workspace, string(newDBPath2))
+	}
+
+	// use the state trie DB that is specified in miscDB
+	{
+		newDBPath := []byte("NEW_STATE_TRIE_DB_PATH")
+		err := node.ChainDB().GetMiscDB().Put(stateTriePathKey, newDBPath)
+		assert.NoError(t, err)
+		stopNode(t, fullNode)
+		_, _, err = newKlaytnNode(t, workspace, validator)
+		assert.Error(t, err, "start failure expected, changed state trie db has no data")
+	}
+}
+
+// checkStatTrieDB writes random values to state trie and checks if the values are stored in expected DB
+func checkStatTrieDB(t *testing.T, fullNode *node_.Node, node *cn.CN, validator *TestAccountType, workspace string, dbPath string) (*node_.Node, *cn.CN) {
+	entries := writeRandomValueToStateTrieDB(t, node.ChainDB().NewBatch(database.StateTrieDB))
+	time.Sleep(10 * time.Second)
+
+	stopNode(t, fullNode) // release DB lock
+	checkIfStoredInDB(t, filepath.Join(node.ChainDB().GetDBConfig().Dir, dbPath), entries)
+	return startNode(t, workspace, validator)
```

Comment:
I think putting `startNode(t, workspace, validator)` on `TestMigration_StartMigrationByMiscDB` function is better.
It is easy to understand the test flow.

Reply:
It is possible to move `stopNode` also?